### PR TITLE
Simulator

### PR DESCRIPTION
**Pull request by GuillaumeDSM**

### CODE REVIEW

File: octobot/community/supabase_backend/community_supabase_client.py [hunk 1]
```diff
@@ -263,10 +264,27 @@ async def fetch_bot_profile_data(self, bot_config_id: str) -> commons_profiles.P
             ].get("minimal_funds", [])
         ] if bot_config[enums.BotConfigKeys.EXCHANGES.value] else []
         profile_data.profile_details.version = bot_config["product_config"][enums.ProfileConfigKeys.VERSION.value]
+        profile_data.trader_simulator.enabled = bot_config.get("simulated", False)
+        if profile_data.trader_simulator.enabled:
+            portfolio = (bot_config.get(
+                enums.BotConfigKeys.OPTIONS.value
+            ) or {}).get("portfolio")
+            usd_like_asset = "USDT"  # todo use dynamic value when exchange is not supporting USDT
```
Comment (Herklos) [inline octobot/community/supabase_backend/community_supabase_client.py:272]: Could we use the strategy reference market if it's an usd like asset?

**Review by Herklos — APPROVED** [review]

File: octobot/community/supabase_backend/community_supabase_client.py (hunk 1 above)
Comment (GuillaumeDSM) [reply octobot/community/supabase_backend/community_supabase_client.py:272]: good idea 👍

File: octobot/community/supabase_backend/community_supabase_client.py [hunk 2]
```diff
@@ -263,10 +265,30 @@ async def fetch_bot_profile_data(self, bot_config_id: str) -> commons_profiles.P
             ].get("minimal_funds", [])
         ] if bot_config[enums.BotConfigKeys.EXCHANGES.value] else []
         profile_data.profile_details.version = bot_config["product_config"][enums.ProfileConfigKeys.VERSION.value]
+        profile_data.trader_simulator.enabled = bot_config.get("simulated", False)
+        if profile_data.trader_simulator.enabled:
+            portfolio = (bot_config.get(
+                enums.BotConfigKeys.OPTIONS.value
+            ) or {}).get("portfolio")
+            if trading_api.is_usd_like_coin(profile_data.trading.reference_market):
+                usd_like_asset = profile_data.trading.reference_market
```
Comment (Herklos) [inline octobot/community/supabase_backend/community_supabase_client.py:274]: :+1:

**Review by Herklos — APPROVED** [review]

**Review by Herklos — APPROVED** [review]

File: octobot/community/authentication.py [hunk 3]
```diff
@@ -364,6 +364,12 @@ async def refresh_selected_bot(self):
             await self.supabase_client.fetch_bot(self.user_account.bot_id)
         )
 
+    async def refresh_selected_bot_if_unset(self):
+        if not self.user_account.has_selected_bot_data():
+            self.user_account.set_selected_bot_raw_data(
+                await self.supabase_client.fetch_bot(self.user_account.bot_id)
```
Comment (Herklos) [inline octobot/community/authentication.py:370]: Should we also check that `self.user_account.bot_id` is not None ?

Comment (GuillaumeDSM) [reply octobot/community/authentication.py:370]: it's initialized in constructor, it should never be None, we are fine :)

Comment (Herklos) [reply octobot/community/authentication.py:370]: Great!

**Review by Herklos — APPROVED** [review]

**Pull request merged**